lib/galaxy/tool_util/linters/inputs.py

```diff
@@ -1,9 +1,22 @@
+
 """This module contains a linting functions for tool inputs."""
-from galaxy.tools.parameters.dynamic_options import filter_types
 from galaxy.util import string_as_bool
 from ._util import is_datasource, is_valid_cheetah_placeholder
 from ..parser.util import _parse_name
 
+FILTER_TYPES = [
```

Comment:
This is a good fix, so +1. But, I'm wondering, would it make sense to make it the caller's responsibility to pass this list to the `lint_inputs` function? The function only needs to check items against this list, and so, I think, it doesn't seem to be the best place where the list is defined. We already list these types in the docstrings for the filter class definitions + the `filter_types` dict - but those are at least in the same module; but here we add this to a different package.

Reply:
OK, so if `lint_inputs` is called dynamically (and from the same tool_util package), I don't think passing the list as argument is a reasonable option. And I can't think of any alternative suggestions.